google/api_core/retry/__init__.py

```diff
@@ -29,6 +28,22 @@
 from .retry_streaming_async import AsyncStreamingRetry
 from .retry_streaming_async import retry_target_stream as retry_target_stream_async
 
+# The following imports are for backwards compatibility with https://github.com/googleapis/python-api-core/blob/4d7d2edee2c108d43deb151e6e0fdceb56b73275/google/api_core/retry.py
+#
+# TODO: Revert these imports on the next major version release (https://github.com/googleapis/python-api-core/issues/576)
+import datetime  # noqa: F401
+import functools  # noqa: F401
+import logging  # noqa: F401
+import random  # noqa: F401
+import sys  # noqa: F401
+import time  # noqa: F401
+import inspect  # noqa: F401
+import warnings  # noqa: F401
+from typing import Any, Callable, TypeVar, TYPE_CHECKING  # noqa: F401
```

Comment:
Searches like [this one on GitHub](https://github.com/search?q=from+google.api_core+import+sys&type=code) and [this one on Google](https://www.google.com/search?q=%22from+google.api_core+import+sys%22+site%3A%3Ahttps%3A%2F%2Fgithub.com&sca_esv=20eb13cd3512c25f&sxsrf=ACQVn09ut9LtpG7Ydx9gcFmRynuHonCbwQ%3A1706559104219&ei=gAa4ZYmJDbjJkPIP4-2xyAs&ved=0ahUKEwiJyceJtIOEAxW4JEQIHeN2DLkQ4dUDCBA&uact=5&oq=%22from+google.api_core+import+sys%22+site%3A%3Ahttps%3A%2F%2Fgithub.com) suggest that there aren't many repos, if any, importing standard packages in non-standard ways, so maybe we're fine to to not expose them here?

Reply:
After a separate discussion, we agreed to not expose standard imports transitively if we don't need them in our implementation. 

I think it would be good if the Python community spelled this out somewhere, but that's a separate issue.